### PR TITLE
Removable Sanitizers

### PR DESCRIPTION
**Pull request by scbedd**
TODO:

- [x] Sanitizers now have an `id`
- [x] Existing behavior of the proxy WRT registering a sanitizer for a session/recording is maintained, but now sanitizers are stored in a central location
- [x] Investigate 2 failing unit tests from JS integration test run, resolve.
  - ~~Getting some assistance from @HarshaNalluru  on this failure, thank you Harsha!~~
  - `2ece462` resolved issues. Glad the tests caught it thanks again Harsha!
- [x] POST Admin/AddSanitizer now returns a body containing the newly added sanitizerId
- [x] GET Admin/Sanitizers is a new route that returns either the session sanitizers (with ids) or specific recordingId based on the presence of `x-recording-id` header
- [x] POST Admin/RemoveSanitizer
- [x] and POST Admin/RemoveSanitizers should _remove_ the given sanitizer Ids from the active session or recording
  - Arguably I should use `DELETE` here, but some clients run into problems sending along a body with a DELETE request, so I'm just going to keep this `POST` and keep it dumb simple.
- [x] Tests for RemoveSanitizer

### CODE REVIEW

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Properties/launchSettings.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "profiles": {
     "Azure.Sdk.Tools.TestProxy": {
       "commandName": "Project",
+      "commandLineArgs": "--storage-location=C:/repo/sdk-for-js",
```
Comment (benbp) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Properties/launchSettings.json:5]: Dev needs to be removed?

**Review by benbp — APPROVED** [review]

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Properties/launchSettings.json (hunk 1 above)
Comment (scbedd) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/Properties/launchSettings.json:5]: Yes absolutely. Thanks for catching it.

**Pull request merged**